### PR TITLE
Redirect to New Workspace Creation with Invalid Policy Id in Workspace URL

### PR DESCRIPTION
**Pull request by mananjadhav**
@timszot Can you please review?

### Details
1. Added redirect to New Workspace page in case the URL contains invalid policy Id
2. Moved navigation in workspace creation to next promise then block
3. Added long duration to the error message

### Fixed Issues

$ #4350 

### Tests
1. Tested with invalid `policyId` in the URL (random numbers and characters)
2. Tested with undefined in the URLs for Policy Id
3. Tested New Workspace creation and then the redirection with this flow
4. Verified existing workspace redirects are not impacted.

### Tested On

- [x] Web
- [x] Mobile Web

#### Web
https://user-images.githubusercontent.com/3069065/128579756-72b72964-5701-436f-91b9-13d997fb2526.mp4

#### Mobile Web
https://user-images.githubusercontent.com/3069065/128579769-e7dc46b7-477c-4a2e-9fd5-16a4bcc1bd04.mp4

### CODE REVIEW

**Comment by mananjadhav** [comment]
@timszot Work done. Few things with this PR.

1. Need help with the message translations.
2. I've added 2 fixes in workspace creation flow too - added `.then` block and `successMessage` translations
3. Error duration is 3500ms, compared to regular 2000ms. Please let me know if 3500ms works.

**Comment by timszot** [comment]
Sorry for the delay here. I'll give this a review and work on getting the translations for you as well, but please give us some time as we're currently [prioritising a company wide internal project](https://expensify.slack.com/archives/C01GTK53T8Q/p1628527713124200).

**Review by timszot — APPROVED** [review]

**Comment by TomatoToaster** [comment]
Hey just a note for when this hits QA stage. We're getting rid of the `Create Workspace` page so these changes will be in effect slightly differently. We are making every instance where you'd see a modal to create a workspace instead just auto-create the workspace with a default name. So here, instead of seeing the modal when the policy in the link doesn't exist, one will just be auto created.

I'd say this doesn't need any further QA when it hits staging and instead the QA will be handled in the other PR where we're changing the `Create Workspace` behavior. Just making a note here so that not seeing a `Create Workspace` modal is NOT treated as a regression for this PR.

**Comment by mananjadhav** [comment]
@TomatoToaster Are you saying that with the other PR, my current PR's modal won't show. One workspace with the default name will be auto created and I don't need to handle it in the current PR, right?

**Comment by TomatoToaster** [comment]
Yep you got it. Nothing left to do on your side 👍🏾

**Comment by mananjadhav** [comment]
@isagoico From QA, we could just check that it doesn't land up to a blank Workspace Card page?

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-08-24.  :confetti_ball:

**Review by marcaaron — COMMENTED** [review]
Noticed a bug on dev related to this PR, but unsure if it's causing any significant issues. What I observed was a failure to create a policy and then the app crashes. Unfortunately, I didn't manage to catch why the policy wasn't created.

Gonna create an issue to make sure we handle this case properly.

File: src/libs/actions/Policy.js [hunk 1]
```diff
@@ -193,16 +194,18 @@ function create(name = '') {
...
                 employeeList: getSimplifiedEmployeeList(response.policy.employeeList),
                 id: response.policyID,
                 type: response.policy.type,
                 name: response.policy.name,
                 role: CONST.POLICY.ROLE.ADMIN,
             });
+        }).then(() => {
             Navigation.dismissModal();
-            Navigation.navigate(ROUTES.getWorkspaceCardRoute(response.policyID));
+            Navigation.navigate(ROUTES.getWorkspaceCardRoute(res.policyID));
```
[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/libs/actions/Policy.js:208]: Okay. Give me a day I'll check. Afaik, Workspace creation was manual with the form when this PR was merged and we did test this.

I'll fix it anyway.

File: src/pages/workspace/WorkspaceSidebar.js [hunk 2]
```diff
@@ -62,9 +64,15 @@ const WorkspaceSidebar = ({translate, isSmallScreenWidth, policy}) => {
         },
     ];
 
-    if (_.isEmpty(policy)) {
-        return null;
-    }
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/pages/workspace/WorkspaceSidebar.js:67]: Okay. I'll check and update.

[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/pages/workspace/WorkspaceSidebar.js:67]: I'll mostly be working on this issue https://github.com/Expensify/App/issues/5185 and it involves working on `WorkspaceSidebar`. I'll update this block of code too when working with this PR. 

Will change `WorkspaceSidebar` to a class component.

[the remark this answers is not in the document]
Comment (TomatoToaster) [reply src/pages/workspace/WorkspaceSidebar.js:67]: Hey just want to note, I actually got rid of the useEffect in one of my PRs when I was editing this component. You won't need to update it!

[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/pages/workspace/WorkspaceSidebar.js:67]: @TomatoToaster  Thanks for the note. :)

File: src/libs/actions/Policy.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/libs/actions/Policy.js:208]: @TomatoToaster Does your change cover this as well? or I'll cover this?

**Pull request merged**